### ISSUE TITLE
Upgrade to GTK3 - Duplicati fails to compile on Debian Testing (Stretch)

### ISSUE THREAD

**Issue by thomassross** [duplicati/duplicati#2389]
I have:
- [x] searched open and closed issues for duplicates

----------------------------------------

### Version info

**Duplicati Version:** 2.0.1.53-2.0.1.53_experimental_2017-03-13
**Operating System:** Debian Linux Testing Stretch (Mono 4.6.2, NuGet 3.5.0.1996)
**Backend:** GTK? Mono? I'm not sure what this means

### Bug description
Duplicati.GUI.TrayIcon fails to build, error:

```
Errors:

/home/thomas/Downloads/duplicati/duplicati-2.0.1.53-2.0.1.53_experimental_2017-03-13/Duplicati.sln (default targets) ->
(Build target) ->
/home/thomas/Downloads/duplicati/duplicati-2.0.1.53-2.0.1.53_experimental_2017-03-13/Duplicati/GUI/Duplicati.GUI.TrayIcon/Duplicati.GUI.TrayIcon.csproj (default targets) ->
/usr/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets (CoreCompile target) ->

	GtkRunner.cs(173,48): error CS0012: The type `GLib.IIcon' is defined in an assembly that is not referenced. Consider adding a reference to assembly `gio-sharp, Version=3.0.0.0, Culture=neutral, PublicKeyToken=35e10195dab3c99f'
```

Build log of relevant target: https://paste.debian.net/plainh/7e237772

### Steps to reproduce
- Download the latest release snapshot
- Extract it
- `cd` in and run `./make`

**Actual result:** It fails to build
**Expected result:** It builds successfully

**Comment by kenkendk** [comment]
I think this means you have to install this package:
```
https://packages.debian.org/sid/libgio2.0-cil-dev
```

Not sure what you have installed previously, but you can try the packages listed here:
https://github.com/duplicati/duplicati/blob/master/Installer/debian/debian/control#L12

**Comment by thomassross** [comment]
Hm... I already have that installed and installing all of the packages in the control file did not help either. Still getting the same error.

**Comment by thomassross** [comment]
I got it to compile. I had to make modifications so that duplicati would use the newest version of appindicator-sharp included in Debian stretch and gtk 3 instead of 2.12.0.0. Here's the patch:

https://paste.debian.net/922861/

**Comment by ts678** [comment]
[Debian “stretch” Release Information](https://www.debian.org/releases/stretch/) says that it's been out of support for years now.
The links here to `paste.debian.net` no longer work. Is this issue still of any interest?
It won't disappear, but if nothing else gets posted, it will move to the `Closed` status.
